DatadogRUM/Sources/Instrumentation/WatchdogTerminations/VendorIdProvider.swift

```diff
@@ -0,0 +1,25 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+#if canImport(UIKit)
+import UIKit
+#endif
+
+/// A type providing the vendor identifier of the device.
+internal protocol VendorIdProvider {
+    /// The vendor identifier of the device.
+    var vendorId: String? { get }
+}
+
+extension UIDevice: VendorIdProvider {
+    var vendorId: String? {
+        let device = UIDevice.current
+        return device.identifierForVendor?.uuidString
+    }
```

Comment:
question/ What do we use `identifierForVendor` for? From [`UIDevice.identifierForVendor` docs](https://developer.apple.com/documentation/uikit/uidevice/1620059-identifierforvendor) it seems to be meant for identifying apps from the same app group, which isn't the case for our SDK.

> The value of this property is the same for apps that come from the same vendor running on the same device. A different value is returned for apps on the same device that come from different vendors, and for apps on different devices regardless of vendor.

I wonder if using it won't flag our SDK as trying to fingerprint the user (like advertisement SDKs do). What alternatives could we consider?

I can see the comment later in the code:
> This value can change when installing test builds using Xcode or when installing an app on a device using ad-hoc distribution.
and I wonder if this isn't a niche use case that we try to support through significant means. I might be just over sceptical here 🙌.

Reply:
Got it 👍. Sounds good if it's already proven in other SDKs 👌.